server/api/accounts.js

```diff
@@ -16,15 +16,24 @@ const register = function (server, serverOptions) {
         path: '/api/accounts',
         options: {
             tags: ['api','accounts'],
+            description: 'Get a paginated list of all accounts (need admin role)',
+            notes: 'This endpoint returns a paginated list of accounts that have been created. This is a much longer detailed description if we have answers to common questions, but I mainly have questions. Whats the difference between accounts and users? When would you want to list all accounts instead of all users? Assuming you need an admin role to use this endpoint? Where is that configured?',
             auth: {
                 scope: 'admin'
             },
             validate: {
                 query: {
-                    sort: Joi.string().default('_id'),
-                    limit: Joi.number().default(20),
-                    page: Joi.number().default(1)
+                    sort: Joi.string().default('_id').description('This is a description for this query parameter. This is obviously a sort but what are the allowed values in the default setup? I\'ve added an example allow() validation which will show up as a pick list in interactive docs.').allow('_id', '_option2', '_option3'),
+                    limit: Joi.number().integer().default(20).description('I\'m assuming this is how many results you want at a time per page? We should mention that just so it\'s clear. It should also have min/max Joi validation which also shows up in docs. I\'ve added an example.').min(1).max(1000),
+                    page: Joi.number().integer().default(1).description('Looks like default is 1 so this is a 1 based index not a zero based index right? We should state stuff like that here.').min(1).max(1000)
```

Comment:
Maybe it's just new to the eyes, but my first impression of the "code" view with these added descriptions is that it feels a bit cluttered.

Reply:
Yep no arguing that it doesn't clutter the code but the metadata has to go somewhere. I was sloppy in this example. We could stick to keeping Joi validation in a cleaner format like this to be more readable and less cluttered looking: https://www.dropbox.com/s/gn9s0o0ih65rq6q/Screenshot%202018-05-24%2019.36.15.png?dl=0